drivers/sensor/tmag5273/tmag5273.h

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright (c) 2023 deveritec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_TMAG5273_CFG_H_
+#define ZEPHYR_DRIVERS_SENSOR_TMAG5273_CFG_H_
+
+#include <stdint.h>
+
+#include <zephyr/device.h>
+
+#include <zephyr/dt-bindings/sensor/tmag5273.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor/tmag5273.h>
+
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/util.h>
```

Comment:
Please change the guard macro to match the filename (i.e. remove the `CFG_` part).

Also, can the additional includes be moved to the driver .c file along with private driver structures?

That way this header declares registers only.